src/auth/hooks.ts

```diff
@@ -268,22 +258,18 @@ export const useAuthSetting = (storage: Storage, options?: UseQueryOptions<AuthS
   return query.data
 }
 
-export const getAuthSetting = async (storage: Storage) => {
-  const authSetting = parseAuthSetting(await storage.getItem(SettingsStorageKeys.Auth))
-  if (isAuthSetting(authSetting)) return authSetting
+export const getAuthSetting = async (storage: YoroiStorage) => {
+  const authSetting = await storage.join('appSettings/').getItem('auth', parseAuthSetting)
+  if (authSetting !== null) return authSetting
   return Promise.reject(new Error('useAuthSetting invalid data'))
 }
 
 const parseAuthSetting = (data: unknown) => {
-  if (!data) return undefined
-  try {
-    return JSON.parse(data as string)
-  } catch (error) {
-    return undefined
-  }
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const isAuthSetting = (data: any): data is 'os' | 'pin' | undefined => ['os', 'pin', undefined].includes(data)
```

Comment:
why is `undefined` a valid auth setting?

Reply:
during the first run / migration, it is, there is no default value